salt/states/idem.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+#
+# Author: Tyler Johnson <tjohnson@saltstack.com>
+#
+
+"""
+Idem Support
+============
+
+This state provides access to idem states
+"""
+from __future__ import absolute_import, print_function, unicode_literals
+
+import logging
+import os
+import sys
+
+import salt.modules.idem
+
+log = logging.getLogger(__name__)
+
+# Function alias to make sure not to shadow built-in's
+__func_alias__ = {"exec_": "exec"}
+__virtualname__ = "idem"
+
+
+def __virtual__():
+    if sys.version_info < (3, 6):
+        return False, "idem only works on python3.6 and later"
+    if not salt.modules.idem.HAS_POP[0]:
+        return salt.modules.idem.HAS_POP
+    return __virtualname__
+
+
+def __init__(opts):
+    if "idem.hub" not in __context__:
+        __context__["idem.hub"] = salt.modules.idem._create_hub()
+
+
+def _get_refs(sources, tree):
+    """
+    Determine where the sls sources are
+    """
+    sls_sources = []
+    SLSs = []
+    for sls in sources:
+        if os.path.isfile(sls):
```

Comment:
I'm not asking for a change, but have you looked at `pathlib` from the stdlib?

Reply:
I'd ask for a change 😂 

Especially since this requires Python 3.6+, there's no reason to *not* use pathlib.